packages/p2p/src/components/buy-sell/popup.jsx

```diff
@@ -13,45 +13,65 @@ import {
-        // eslint-disable-next-line no-console
-        console.log(this.state);
-        // eslint-disable-next-line no-console
-        console.log(formik_vars);
-        setSubmitting(false);
+    handleSubmit = async (values, { setSubmitting }) => {
+        const { ad } = this.props;
+        
+        const order = await MockWS({ p2p_order_create: 1, amount: values.send, offer_id: ad.offer_id });
+
+        if (!order.error) {
+            const order_info = await MockWS({ p2p_order_info: 1, order_id: order.order_id });
+            this.props.onConfirm(order_info);
+            setSubmitting(false);
+            this.props.onCancel();
+        } else {
+            // TODO: [p2p-handle-error] handle error on order creation
+            setSubmitting(false);
+        }
+
     }
 
     getInitialValues = (is_buy) => {
         const { ad } = this.props;
-        const amount_currency = ad.min_transaction;
-        const amount_asset = ad.min_transaction / ad.fix_price;
+
+        const amount_asset = +(ad.min_transaction / ad.price).toFixed(ad.transaction_currency_decimals);
+
         const buy_initial_values = {
-            initial_send    : amount_currency,
             initial_receive : amount_asset,
-            send_currency   : ad.currency,
-            receive_currency: ad.asset,
+            initial_send    : ad.min_transaction,
+            receive_currency: ad.offer_currency,
+            receive_decimals: ad.offer_currency_decimals,
+            send_currency   : ad.transaction_currency,
+            send_decimals   : ad.transaction_currency_decimals,
         };
         const sell_initial_values = {
+            initial_receive : ad.min_transaction,
             initial_send    : amount_asset,
-            initial_receive : amount_currency,
-            send_currency   : ad.asset,
-            receive_currency: ad.currency,
+            receive_currency: ad.transaction_currency,
+            receive_decimals: ad.transaction_currency_decimals,
+            send_currency   : ad.offer_currency,
+            send_decimals   : ad.offer_currency_decimals,
         };
 
         return is_buy ? buy_initial_values : sell_initial_values;
     }
 
     calculateReceiveAmount = (send_amount, is_buy) => {
         const { ad } = this.props;
-        return is_buy ? send_amount / ad.fix_price : send_amount * ad.fix_price;
+        return is_buy ?
+            +(send_amount / ad.price).toFixed(ad.transaction_currency_decimals)
+            :
+            +(send_amount * ad.price).toFixed(ad.transaction_currency_decimals);
     };
 
     calculateSendAmount = (receive_amount, is_buy) => {
         const { ad } = this.props;
-        return is_buy ? receive_amount * ad.fix_price : receive_amount / ad.fix_price;
+        return is_buy ?
+            +(receive_amount * ad.price).toFixed(ad.transaction_currency_decimals)
+            :
+            +(receive_amount / ad.price).toFixed(ad.transaction_currency_decimals);
```

Comment:
Why is the plus sign there? If it's meant to coerce to number, it would need to be on the send_amount and ad.price themselves, if I'm not mistaken. And same for the lines below.

Reply:
.toFixed changed it to a string, the `+` changed it back to a number

Reply:
Got it. Can we put parentheses around the whole thing then like:

```+((send_amount / ad.price).toFixed(ad.transaction_currency_decimals))```